codegen/masm/src/codegen.rs

```diff
@@ -136,111 +153,168 @@ impl<'ast> CodeGenerator<'ast> {
 
     /// Emits code for the procedure `cache_z_exp`.
     ///
-    /// The emitted procedure computes a `z**exp` for each periodic column, these values are later
-    /// on used to evaluate the polynomial of each periodic column.
+    /// The procedure computes and caches the necessary exponentiation of `z`. These values are
+    /// later on used to evaluate each periodic column polynomial and the constraint divisor.
     ///
-    /// The emitted code is optimized to performn the fewest number of exponentiations, this is
-    /// achieved by observing that periodic columns and trace length are both power-of-two in
-    /// length, since the exponent is defined as `exp = trace_len / periodic_column_len`, all
-    /// exponents are themselves powers-of-two.
+    /// This procedure exists because the VM doesn't have native instructions for exponentiation of
+    /// quadratic extension elements, and this is an expensive operation.
     ///
-    /// The algorithm computes the exponents from smallest to largest, using the previous result as
-    /// a cache value.
+    /// The generated code is optimized to perform the fewest number of exponentiations, this is
+    /// achieved by observing that periodic columns and trace length are both powers-of-two, since
+    /// the exponent is defined as `exp = trace_len / periodic_column_len`, all exponents are
+    /// themselves powers-of-two. This allows the results to be computed from smallest to largest,
+    /// re-using the intermediary values.
```

Comment:
I think this is the answer to my question above. Basically, we compute `z^n` from `z^m`, where `n` and `m` are two periodic lengths sorted in increasing order, using `log2(n-m)` multiplications?
Can we add a comment to the generated procedures to clarify this (or the correct reason if the above is wrong)?

Reply:
> I think this is the answer to my question above. Basically, we compute z^n from z^m, where n and m are two periodic lengths sorted in increasing order, using log2(n-m) multiplications?

Exactly.

> Can we add a comment to the generated procedures to clarify this (or the correct reason if the above is wrong)?

I updated the comments of the generated code, let me know if that is enough (I have done small changes, I want to know if you want a more verbose text)

Reply:
Looks good to me, thank you!